packages/client/src/components/SafeSettings.js

```diff
@@ -349,6 +351,142 @@ const AddSafeOwner = ({ web3, onCancel, onNext, safeOwners }) => {
   );
 };
 
+const AddVault = ({ onCancel, onNext }) => {
+  const [contractName, setContractName] = useState("");
+  const isFormValid = contractName?.trim().length > 0;
+
+  const onNextClick = () => {
+    onNext({
+      contractName,
+    });
+  };
+
+  const nextButtonClasses = [
+    "button flex-1 p-4",
+    isFormValid ? "is-link" : "is-light is-disabled",
+  ];
+
+  const coinTypes = Object.entries(COIN_TYPE_TO_META).map((type) => ({
+    itemValue: type[0],
+    displayText: type[1].displayName,
+  }));
+
+  const [coinType, setCoinType] = useState(coinTypes[0].displayText);
+
+  useEffect(() => {
```

Comment:
I don't see a user interface for the state contractName, guessing the purpose here is to project coingType to contractName. In this case can we pass the cointype instead of the contractName to the event handler (onAddVaultSubmit)? That way we can optimize out the extra state and useeffect rendering computation.

Reply:
Will update, thank you!

Reply:
Updated!